exe/task.cc

```diff
@@ -137,6 +148,161 @@ static map<string, string> generate_environment_from(lth_jc::JsonContainer &inpu
     return env_vars;
 }
 
+// TODO: PCP-769 added routines start here!
+
+// Take out common permissions.
+fs::perms TASK_FILE_RW_PERMS = fs::owner_read | fs::owner_write | fs::group_read | fs::group_write;
+
+// Creates the SYSTEM_PREFIX/pxp-agent/tasks-cache/<sha> directory,
+// and ensures that its permissions are only readable/writable by
+// the PXP agent owner/group. Returns the path to this directory.
+// Note that the last modified time of the directory is updated.
+/*static*/ fs::path create_tasks_cache_dir(const string &sha256) {
+    auto tasks_cache_dir = fs::path(SYSTEM_PREFIX()) / "pxp-agent" / "tasks-cache" / sha256;
+    try {
+      fs::create_directories(tasks_cache_dir);
+      fs::last_write_time(tasks_cache_dir, time(nullptr)); 
+      fs::permissions(tasks_cache_dir, TASK_FILE_RW_PERMS);
+    } catch (const fs::filesystem_error& e) {
+      // TODO: Throw exception here containing the error message specified in "e". Could be
+      // directory was not created, its last modified time was not set, or the permissions
+      // setting failed. Should this be a PXP-error?
+    }
+    return tasks_cache_dir;
+}
+
+// Computes the sha256 of the file denoted by path. Assumes that
+// the file designated by "path" exists. Can switch to openssl/evp,
+// as described on https://www.openssl.org/docs/man1.1.0/crypto/EVP_DigestInit.html
+// too for a higher level, more flexible interface if necessary.
+/*static*/ string calculate_sha256(const string &path) {
+    const streamsize CHUNK_SIZE = 2 << 14;
+    char buffer[CHUNK_SIZE];
+    unsigned char md[SHA256_DIGEST_LENGTH];
+  
+    SHA256_CTX ctx;
+    SHA256_Init(&ctx);
+    boost::nowide::ifstream ifs(path);
+    while (!ifs.eof()) {
```

Comment:
Usually worth using `while (ifs)`, then check that `ifs.eof()` afterwards. It can also be unreadable for an error, and you probably want to handle that.